### PR TITLE
Asymmetric SecurityAlgorithms have entries for checking minimum key sizes

### PR DESCRIPTION
**Pull request by brentschmaltz**
…-extensions-for-dotnet/issues/1236

Make sure our defined asymmetric SecurityAlgorithms have entries for checking key sizes

### CODE REVIEW

File: src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs [hunk 1]
```diff
@@ -36,6 +38,76 @@ namespace Microsoft.IdentityModel.Tokens
...
+            SecurityAlgorithms.RsaSha384Signature,
+            SecurityAlgorithms.RsaSha512Signature
+        };
+
+        internal static ICollection<string> RsaPssSigningAlgorithms = new Collection<string>
+        {
+            SecurityAlgorithms.RsaSsaPssSha256,
+            SecurityAlgorithms.RsaSsaPssSha384,
+            SecurityAlgorithms.RsaSsaPssSha512,
+            SecurityAlgorithms.RsaSsaPssSha256Signature,
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs:83]: Should this one be added to the supported algorithms wiki page?

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/wiki/Supported-Algorithms

File: src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs [hunk 2]
```diff
@@ -36,6 +38,76 @@ namespace Microsoft.IdentityModel.Tokens
     /// </summary>
     internal static class SupportedAlgorithms
     {
+        internal static ICollection<string> EcdsaSigningAlgorithms = new Collection<string>
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs]: nit: These could all be IReadOnlyCollection

File: src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs [hunk 3]
```diff
@@ -140,7 +189,10 @@ internal static bool IsSupportedKeyWrapAlgorithm(string algorithm, SecurityKey k
             if (string.IsNullOrEmpty(algorithm))
                 return false;
 
-            if (algorithm.Equals(SecurityAlgorithms.RsaPKCS1, StringComparison.Ordinal)
+            if (key.KeySize < 2048)
+                return false;
+
+            if (   algorithm.Equals(SecurityAlgorithms.RsaPKCS1, StringComparison.Ordinal)
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs:195]: nit: extra whitespace

File: src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs [hunk 4]
```diff
@@ -101,35 +173,12 @@ internal static bool IsSupportedAuthenticatedEncryptionAlgorithm(string algorith
...
-                case SecurityAlgorithms.Sha512Digest:
-                    return true;
-
-                default:
-                    return false;
-            }
+            return HashAlgorithms.Contains(algorithm);
         }
 
         internal static bool IsSupportedKeyWrapAlgorithm(string algorithm, SecurityKey key)
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs:184]: out of scope: This is name is not completely honest. It is is only RSA key wrap

**Review by keegan-caruso — APPROVED** [review]

File: src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs (hunk 1 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs:83]: updated wiki

File: src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs (hunk 3 above)
Comment (GeoK) [inline src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs:195]: nit: spacing

**Review by GeoK — APPROVED** [review]

File: src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs (hunk 3 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs:195]: I wanted to have the 'algorithm' variables line up...

**Pull request merged**